### DISCUSSION TITLE
New version 13.0 Filament runout sensor

### DISCUSSION THREAD

**Discussion by ulivero** [Jyers/Marlin#423] [Q&A] [answered]
Hi i have a bltouch and filament runout sensor , when start the print , the sensor detects out of filament and pause the print and unload the filament , when filament is present , theres an option inverted of this firmware?
Thanks in advance

**Comment by Jyers** [comment, marked as answer]
Unfortunately there are two possible pin states from filament runout and there is no way I can make this accessible from the on screen menus. If your able to compile your own firmware, its a simple fix in configuration.h Just change `#define FIL_RUNOUT_STATE LOW` to `#define FIL_RUNOUT_STATE HIGH`

**Reply by ulivero** [reply]
Hi Jyers thanks for the response , where i get the file to compile again , and i have to change all to all working or only the low to high?

**Reply by Cal-LF** [reply]
Low to high should get it going.
Config templates are in the Configuration Files - E3V2 Templates

**Reply by ulivero** [reply]
im sorry Jyers las Q , where are the templates in this web, thanks for all

**Reply by ulivero** [reply]
im sorry how copy to compile the templates i found it but im stuck there

**Reply by ulivero** [reply]
Thank you so much ALL WORKING NOW

**Comment by Cal-LF** [comment]
Have you set the runout length? If it is set to 0, it will trip. 7 is a recommended number. I have mine set to 10,

**Comment by ulivero** [comment]
Yes i put in 0 and then in 7 , start the print and detects filament runout with filament inside

**Reply by Jyers** [reply]
I'm assuming you're using a switch based sensor, which means that like I said, it is probably reversed as to the default
